packages/delegation-process/src/services/readModelService.ts

```diff
@@ -246,6 +249,166 @@ export function readModelServiceBuilder(
+            "data.delegateId": filters.delegateId,
+          } satisfies ReadModelFilter<Delegation>,
+        },
+        {
+          $lookup: {
+            from: "tenants",
+            localField: "data.delegatorId",
+            foreignField: "data.id",
+            as: "delegator",
+          },
+        },
+        {
+          $unwind: "$delegator",
+        },
+        ...(filters.delegatorName
+          ? [
+              {
+                $match: {
+                  "delegator.data.name": {
+                    $regex: ReadModelRepository.escapeRegExp(
+                      filters.delegatorName
+                    ),
+                    $options: "i",
+                  },
+                },
+              },
+            ]
+          : []),
+        {
+          $lookup: {
+            from: "eservices",
+            localField: "data.eserviceId",
+            foreignField: "data.id",
+            as: "eservice",
+          },
+        },
+        {
+          $unwind: "$eservice",
+        },
+        {
+          $project: {
+            _id: 0,
+            consumerId: "$delegator.data.id",
+            consumerName: "$delegator.data.name",
+            eserviceId: "$eservice.data.id",
+            producerId: "$eservice.data.producerId",
+          },
+        },
+      ];
+
+      const data = await delegations
+        .aggregate(aggregationPipeline, { allowDiskUse: true })
+        .toArray();
+
+      const result = z
+        .array(
+          z.object({
+            consumerId: z.string(),
+            consumerName: z.string(),
+            eserviceId: z.string(),
+            producerId: z.string(),
+          })
+        )
+        .safeParse(data);
+
+      if (!result.success) {
+        throw genericInternalError(
+          `Unable to parse compact delegation tenants: result ${JSON.stringify(
+            result
+          )} - data ${JSON.stringify(data)}`
+        );
+      }
+
+      if (result.data.length === 0) {
+        return {
+          results: [],
+          pagination: {
+            offset: filters.offset,
+            limit: filters.limit,
+            totalCount: 0,
+          },
+        };
+      }
+
+      const activeDelegators = await getDelegatorsWithActiveAgreements(
+        result.data
+      );
+
+      const filteredDelegators = Array.from(
+        result.data
+          .reduce((map, delegation) => {
+            if (
+              activeDelegators.includes(delegation.consumerId) &&
+              !map.has(delegation.consumerId)
+            ) {
+              map.set(delegation.consumerId, {
+                id: delegation.consumerId,
+                name: delegation.consumerName,
+              });
+            }
+            return map;
+          }, new Map<string, delegationApi.CompactTenant>())
+          .values()
+      );
+
+      return {
+        results: filteredDelegators.slice(
+          filters.offset,
+          filters.offset + filters.limit
+        ),
```

Comment:
we are not using `offset` and `limit` as condition on the mongo query? could this cause performance issue?

Reply:
> we are not using `offset` and `limit` as condition on the mongo query? could this cause performance issue?

I couldn't include those in the query since there're two different queries being done here because I didn't want to include `agreements` in the lookup as that collection is very large and could cause problems; in this case `limit` and `offset` are applied on the result of those queries

Reply:
> I would try to make this one query and apply offset and limit in the Mongo query - to make this more efficient maybe I could propose something like:
> 
>     * start from delegations, look for active delegations with delegate the requester tenant
> 
>     * Look into agreements for each delegation, looking for an active agreement that has consumerId equal to the delegator
> 
>     * lookup on tenants for the delegator, so that you can apply the name filter

How would you implement the second point without a lookup on the agreements collection? Since we had [this problem](https://pagopa.atlassian.net/browse/PIN-5104) I was trying to avoid doing a lookup on that collection

Reply:
There are other places in the app where we perform lookups on agreements by consumerId or eserviceId, for example [here](https://vscode.dev/github/pagopa/interop-be-monorepo/blob/feature/incaricato/packages/tenant-process/src/services/readModelService.ts#L278) and [here](https://vscode.dev/github/pagopa/interop-be-monorepo/blob/feature/incaricato/packages/catalog-process/src/services/readModelService.ts#L403-L404) - I would proceed with the same approach and perform the lookup - unless we already know for sure it's causing a problem also in these other cases. It should not be too intense since we are fixing both consumerId and eserviceId in this case, resulting in a single active agreement for each consumer/eservice.

Reply:
Refactored [here](https://github.com/pagopa/interop-be-monorepo/pull/1295/commits/d349c98fa7a93b1c0f5baeba4c1b0841be41a298) with the addition of the lookup